ToolBox/GoogleDownloader/GoogleDataDownloader.cs

```diff
@@ -28,37 +30,101 @@ namespace QuantConnect.ToolBox.GoogleDownloader
     public class GoogleDataDownloader : IDataDownloader
     {
         // q = SYMBOL
+        // startdate = Mon+D%2C+YYYY, e.g. Jan+1%2C+2005
+        // enddate = Mon+D%2C+YYYY, e.g. Sep+29%2C+2016
+        private const string UrlPrototypeDaily = @"http://www.google.com/finance/historical?q={0}&startdate={1}&enddate={2}&output=csv";
+        private enum ConvertMonths
+        {
+            Jan = 1,
+            Feb = 2,
+            Mar = 3,
+            Apr = 4,
+            May = 5,
+            Jun = 6,
+            Jul = 7,
+            Aug = 8,
+            Sep = 9,
+            Oct = 10,
+            Nov = 11,
+            Dec = 12
+        };
+
+        private IEnumerable<BaseData> GetDaily(Symbol symbol, Resolution resolution, DateTime startUtc, DateTime endUtc)
+        {
+            var startdate = ((ConvertMonths)startUtc.Month).ToString()
+                + @"+" + startUtc.Day.ToString()
+                + @"%2C+" + startUtc.Year.ToString();
+            var enddate = ((ConvertMonths)endUtc.Month).ToString()
+                + @"+" + endUtc.Day.ToString()
+                + @"%2C+" + endUtc.Year.ToString();
+
+            // Create the Google formatted URL.
+            var url = string.Format(UrlPrototypeDaily, symbol.Value, startdate, enddate);
+
+            // Download the data from Google.
+            string[] lines;
+            using (var client = new WebClient())
+            {
+                var data = client.DownloadString(url);
+                lines = data.Split('\n');
+            }
+
+            // first line is header
+            var currentLine = 1;
+
+            while (currentLine < lines.Length - 1)
+            {
+                // Format: Date,Open,High,Low,Close,Volume
+                var columns = lines[currentLine].Split(',');
+
+                // date format: DD-Mon-YY, e.g. 27-Sep-16
+                var DMY = columns[0].Split('-');
+
+                // date = 20160927
+                var day = DMY[0].ToInt32();
+                var month = (int)Enum.Parse(typeof(ConvertMonths), DMY[1]);
+                var year = (DMY[2].ToInt32() > 70) ? 1900 + DMY[2].ToInt32() : 2000 + DMY[2].ToInt32();
+                var time = new DateTime(year, month, day, 0, 0, 0);
+
+                // occasionally, the columns will have a '-' instead of a proper value
+                List<Decimal?> ohlc = new List<Decimal?>()
+                {
+                    columns[1] != "-" ? (Decimal?)columns[1].ToDecimal() : null,
```

Comment:
Could you please make this log an error message when a null appears? Log.Error() with the date and price that google returned so consumer knows its an invalid point.